### DISCUSSION TITLE
Reservations and tags linking them

### DISCUSSION THREAD

**Discussion by bordera-randy** [Azure/ipam#226] [Q&A] [answered]
I am working to get a reservation to work with the tag linking it to an azure vnet 
I am able to create the reservation and get the X-IPAM-RES-ID and i add X-IPAM-RES-ID as a tag to the azure vnet

However I am not seeing the reservation link to the vnet, I checked the Q&A and documentation but i was not able to find an answer there 

i let this sit overnight to see if it was a timing issue but still see no change 

![image](https://github.com/Azure/ipam/assets/48829240/e01b5d90-8893-43fd-a3ca-f9f1dc884ed6)

![image](https://github.com/Azure/ipam/assets/48829240/b7bfb5a6-df0c-4ac8-833c-a339667924f2)

**Comment by bordera-randy** [comment]
additional info, it looks like the ipam is not seeing the vnet

**Reply by DCMattyG** [reply]
Hey there @bordera-randy, appreciate you reaching out on this issue!

Can you help me understand how you deploy the Azure IPAM solution? I just ant to see if you used the Deployment script as-is, or if there were any modifications or perhaps a custom deployment method you used.

By default, the `deploy.ps1` script will create the required Service Principal and assigned it as **Reader** on the Root Management Group. This will give it visibility to all of the Virtual Networks within your Azure Tenant, hence why I want to verify how this was deployed given you mentioned above that the IPAM tool isn't seeing the newly created Virtual Network.

**Reply by bordera-randy** [reply]
@DCMattyG  we used the standard deployment script and have ensured the service principal has read access on the subscription level for all the subscriptions

if possible we can look at it on a screen share to answer any questions

**Reply by DCMattyG** [reply]
Happy to @bordera-randy, simply drop me an email at Matthew.Garrett@microsoft.com and we can coordinate a time to look at this together.

**Reply by bordera-randy** [reply]
@DCMattyG  thanks! email sent

**Reply by bordera-randy** [reply, marked as answer]
This was resolved and found to be an issue with the azcontext and a excluded subscription
below I have included the powershell script used to reserve the CIDR and create the resource group and vnet in Azure. 


```
Set-AzContext "SUBID_HERE"

$size                   = "23"
$desc                  = "OPTIONAL_DESCRIPTION"
$vnet                   = "VNET_NAME_HERE"
$vnetresourcegroup      = "RESOURCE_GROUP_NAME_HERE"
$location               = "AZURE_LOCATION_HERE"

$space  = "SPACE_NAME_HERE"
$blocks = @("BLOCK_NAME_HERE")
$reverse_search = "false"
$smallest_cidr  = "True"

$engineClientId = 'ENGINE_CLIENTID_HERE'
$accessToken = ConvertTo-SecureString (Get-AzAccessToken -ResourceUrl api://$engineClientId).Token -AsPlainText -Force
$requestUrl = "https://IPAM_URL_HERE/api/spaces/$space/blocks/$blocks/reservations"

$body = @{
  "blocks" = $blocks
  "size"   = $size
  "desc"   = $desc
  "reverse_search" = $reverse_search
  "smallest_cidr"  = $smallest_cidr
} | ConvertTo-Json

$headers = @{
  'Accept' = 'application/json'
  'Content-Type' = 'application/json'
}
write-host "INFO: Creating IPAM Reservation"
$response = Invoke-RestMethod -Method 'POST' -Uri $requestUrl -Headers $headers -Body $body -Authentication Bearer -token $accessToken

$Tag = @{
    "X-IPAM-RES-ID" = $response.tag.'X-IPAM-RES-ID'
}

write-host "INFO: Creating Resource Group"
$rg = New-AzResourceGroup -Name $vnetresourcegroup -Location $location -Tag $Tag
write-host "INFO: Creating vNet with CIDR $($response.cidr)"
$vnetinfo = New-AzVirtualNetwork -Name $vnet -ResourceGroupName $vnetresourcegroup -Location $location -AddressPrefix $response.cidr -Tag $Tag
```
